sejas/gutenberg
# SiteTitle: Add bold and italic controls

## What?

Add two control buttons to the SiteTitle toolbar to change the `Bold` and `Italic` text styles.

## Why?

The `SiteTitle` style options are difficult to access. I propose adding the bold and italic buttons to the toolbar allowing users to edit the text style easily.
In addition, we improve the accessibility of the block by allowing the shortcuts `Ctrl + B` to make the text bold and `Ctrl + I` to make it italic.

## How?

I've added two new Toolbar buttons to the SiteTitle block. I'm relaying to the `attribtues.style` as the source of truth of the bold and style. For this reason, the new buttons and the Appearance Selector show the same state working together in sync. Check the screencast to see it in action.

<img width="174" alt="Screenshot 2022-12-13 at 01 48 50" src="https://user-images.githubusercontent.com/779993/207206095-e2d938a9-383e-4c82-be58-d4e8b6d2b0d5.png">

## Testing Instructions




1. Go to Appearance > Editor.
2. Click on `Edit`.
3. Select your `SiteTitle` block.
4. Observe that there are two new buttons on the Toolbar to convert the text to Bold and Italic, respectively.
5. Click on the Bold button.
6. Observe the text transforms to bold.
7. Click on the Italic button.
8. Observe the text transforms to italic.
9. Observe that the Appearance select in the inner settings has changed too. The toolbar and the select are in sync.
10. Save the template.
11. Observe the Frontend of the site shows the correct styles.


### Testing Instructions for Keyboard

1. Select the `SiteTitle` block
8. Press `Ctrl + B` or `⌘ + B`
12. Observe the text gets bold.
13. Press `Ctrl + I` or `⌘ + I`
14. Observe the text gets italic.

## Screenshots or screencast 

| **Before** | **After** |
|---|---|
| <img width="1178" alt="site-title-before" src="https://user-images.githubusercontent.com/779993/207203618-c669f32a-7566-4b17-8190-e9c221e327e9.png"> | <img width="1216" alt="site-title-after" src="https://user-images.githubusercontent.com/779993/207203652-68542345-2d93-41f2-8bed-88ddaf0c669d.png"> |



https://user-images.githubusercontent.com/779993/207203733-3d1b898b-f989-4569-8cda-5f29f9a2b482.mp4
